create public release

arjun-krishna1 opened the pull request.

alexccl commented on README.md:
```diff
@@ -1,2 +1,87 @@
+### Historical Data
+
+- `test_cases.json` - 1,000 historical input/output examples
+- `test_cases.csv` - Same data in CSV format for easy analysis
```
this does not appear to be included yet

Melbourneandrew replied:
Confusing. Just give CSV.

alexccl commented on README.md:
```diff
@@ -1,2 +1,87 @@
+- **Average error**: Mean absolute difference from expected outputs
+- **Score**: Lower is better (combines accuracy and precision)
+
+## ubmission
```
typo

Melbourneandrew commented on PRD.md:
```diff
@@ -0,0 +1,73 @@
+
+Your replica will be tested against 1,000 historical reimbursement cases
+
+It will then run in a shadow deployment on 5,000 live reimbursement requests
```
Maybe we can elaborate on a "shadow deployment". Also, I was under the impression that they would be running the code on the 5k samples and submitting them for grading.

Melbourneandrew commented on PRD.md:
```diff
@@ -0,0 +1,73 @@
+- How closely your system's outputs match the legacy system
+- Your ability to explain discrepancies or special-case logic
+- Clear documentation of all inferred business rules
+- High performance (sub-millisecond response time)
```
Performance criteria? I think the only success criteria should be how close the outputs match.

arjun-krishna1 replied:
Removed the other success criteria @Melbourneandrew

Melbourneandrew commented on run.sh.template:
```diff
@@ -0,0 +1,21 @@
+
+# Example implementations (choose one and modify):
+
+# Option 1: Python implementation
```
Instead of "Options" maybe we call them "Examples" to emphasize that they could use any language that they want.

Melbourneandrew commented on PRD.md:
```diff
@@ -0,0 +1,73 @@
+- Produces the same numeric output as the legacy system
+- Matches the system's behavior across a wide variety of scenarios—including edge cases and likely bugs
+- Runs without any external services (in-memory only)
+- Completes each reimbursement calculation in under 1 millisecond
```
I am against this requirement. No way for us to verify.

arjun-krishna1 replied:
Removed @Melbourneandrew

arjun-krishna1 commented on test_cases.json:
@alexccl / @Alex-Downey  updated the Validation section of the README to clarify that there is a validation set of 5000 that's not included in test cases.json

```
Your submission will be tested against 5,000 additional cases that are not included in test_cases.json.
```

arjun-krishna1 commented:
Removed @Melbourneandrew , you are correct there is no shadow deployment
There is the public test set with 1000 that they have been given
And 5000 private test cases that they can't see but we test on

https://github.com/8090-inc/black-box-challenge/pull/1#discussion_r2110386429

arjun-krishna1 commented:
Removed CSV reference @Melbourneandrew , there is only a JSON
So kept that reference and removed the other one:
https://github.com/8090-inc/black-box-challenge/pull/1#discussion_r2110388950

Ro-Fo commented:
i guess since the start is on Saturday at 10AM PT you should make this private ;)
